packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -416,193 +416,204 @@ export function getValidatorApi({
     }
   };
 
-  const produceBlockV3: ServerApi<routes.validator.Api>["produceBlockV3"] = async function produceBlockV3(
-    slot,
-    randaoReveal,
-    graffiti,
-    // TODO deneb: skip randao verification
-    _skipRandaoVerification?: boolean,
-    {feeRecipient, builderSelection, strictFeeRecipientCheck}: routes.validator.ExtraProduceBlockOps = {}
-  ) {
-    notWhileSyncing();
-    await waitForSlot(slot); // Must never request for a future slot > currentSlot
-
-    // Process the queued attestations in the forkchoice for correct head estimation
-    // forkChoice.updateTime() might have already been called by the onSlot clock
-    // handler, in which case this should just return.
-    chain.forkChoice.updateTime(slot);
-    chain.recomputeForkChoiceHead();
-
-    const fork = config.getForkName(slot);
-    // set some sensible opts
-    builderSelection = builderSelection ?? routes.validator.BuilderSelection.MaxProfit;
-    const isBuilderEnabled =
-      ForkSeq[fork] >= ForkSeq.bellatrix &&
-      chain.executionBuilder !== undefined &&
-      builderSelection !== routes.validator.BuilderSelection.ExecutionOnly;
-
-    logger.verbose("Assembling block with produceBlockV3 ", {
-      fork,
-      builderSelection,
+  const produceEngineOrBuilderBlock: ServerApi<routes.validator.Api>["produceBlockV3"] =
```

Comment:
changes in `produceEngineOrBuilderBlock` (old `produceBlockV3`) are because of formatting changes introduced because of name length changes may be, but this function doesn't essentially change and produceBlockV3 is redefined which uses now this  and handles blinding of local payload if requested